src/api/EpiData.ts

```diff
@@ -147,26 +162,50 @@ export function loadDataSet(
   url.searchParams.set('format', 'json');
   return fetchImpl<Record<string, unknown>[]>(url)
     .then((res) => {
-      const data = loadEpidata(title, res, columns, columnRenamings, { _endpoint: endpoint, ...params });
-      if (data.datasets.length == 0) {
+      try {
+        const data = loadEpidata(title, res, columns, columnRenamings, { _endpoint: endpoint, ...params });
+        if (data.datasets.length == 0) {
+          return UIkit.modal
+            .alert(
+              `
+        <div class="uk-alert uk-alert-error">
+          <a href="${url.href}">API Link</a> returned no data, which suggests that the API has no available information for the selected location.
+        </div>`,
+            )
+            .then(() => null);
+        }
+        return data;
+      } catch (error) {
+        console.warn('failed loading data', error);
+        // EpiData API error - JSON with "message" property
+        if ('message' in res) {
+          return UIkit.modal
+            .alert(
+              `
+          <div class="uk-alert uk-alert-error">
+            Failed to fetch API data from <a href="${url.href}">API Link</a>:<br/><i>${res['message']}</i>
```

Comment:
can we make the three versions of this message ever-so-slightly different (for ease of future debugging)?
```suggestion
            Failed to fetch API data from <a href="${url.href}">API Link</a>: (code f01)<br/><i>${res['message']}</i>
```